### ISSUE TITLE
playground name validations

### ISSUE THREAD

**Issue by dimitriirtimid** [local-motion/product#400]
The playground name must match the following criteria:
- min 2 characters
- max 40 characters
- allowed characters: a-z A-Z 0-9 space - & ! '
- no consecutive spaces
- forced trimming of the names

![image](https://user-images.githubusercontent.com/24209923/58983872-05b5d880-87d8-11e9-833d-77022ea54d6f.png)

**Comment by dimitriirtimid** [comment]
@maxvisser Can you please validate these specs?

**Comment by maxvisser** [comment]
Yes looks great. Also thought a bit more about a name being unique; as it is currently uncertain if there will be a copy of the application for new use case; I can see workspaces having not unique names in some other use cases. As we currently can't really grasp the full scope of these other use cases and if this is really needed; I think we should leave it as it is now (unique names).

**Comment by dimitriirtimid** [comment]
Agreed. In the end we may end up with prefixing the name with the 'campaign' name (like smokefree playgrounds) or having dedicated urls / domains per campaign (like we have currently). 
The uniqueness requirement is mainly in order to not confuse the users. The system has no problem with non-unique names, except for the url encoding. But we can always adapt this at a later stage.

Note that I added one more constraint: There can be no consecutive spaces in the playground name.

Closing this issue now as the logic has already been completed and merged in
